pages/products.page.ts

```diff
@@ -0,0 +1,60 @@
+import { Page as PlaywrightPage, Locator, expect } from '@playwright/test';
+import Page from './page';
+
+class ProductsPage extends Page {
+    public locators: { [key: string]: Locator };
+
+    constructor(page: PlaywrightPage) {
+        super(page);
+        this.locators = {
+            produtsList: this.page.locator('div[class*="UnitCard_cardWrapper"]>a'),
+            productFilterItem: this.page.locator('div[class*="ResetFilters_selectedCategory"]'),
+            dropdownArrow: this.page.locator('#__next > div > main > div > div > div[class*="Filters_leftContainer"] > div > div[class*="Filters_filtersWrapper"] > div[class*="Services_wrapper"] > div:nth-child(1) > div[class*="ServiceCategory_svgContainer"]'),
+            unitsContainer: this.page.locator('div[class*="MapPagination_units_container"]'),
+            constructionsCheckBox: this.page.locator('label', {hasText: 'Будівельні'}),
+            othersCheckBox: this.page.locator('label', {hasText: 'Інші'}),
+            searchInput: this.page.getByTestId('searchInput')
+    }
+    }
+
+    async clickFirstProduct() {
+        if(await this.locators.produtsList.first().isVisible()) {
+            await this.locators.produtsList.first().click();
+        }else {}
+    }
+
+    async checkFilters(unitName: string) {
+        this.locators.dropdownCheckBox = this.page.locator('label', { hasText: unitName });
+        if (!(await this.locators.dropdownCheckBox.isVisible())) {
+            await this.locators.dropdownArrow.click()
+        }
+        await expect(this.locators.dropdownCheckBox).toBeChecked()
+    }
+
+    async checkProductsFilter() {
+        await expect(this.locators.productFilterItem).toBeVisible();
+    }
+
+    async clickOnDropdownArrow() {
+        await this.locators.dropdownArrow.click();
+    }
+
+    async checkCategoriesCheckboxes() {
```

Comment:
Bad naming. You are verifying that the checkbox is checked. Checking the checkbox means actual click action on it.